client/python/gradio_client/client.py

```diff
@@ -1015,18 +1012,19 @@ def _download_file(
         if isinstance(x, str):
             file_name = utils.decode_base64_to_file(x, dir=save_dir).name
         elif isinstance(x, dict):
-            if x.get("is_file"):
-                filepath = x.get("name")
-                assert filepath is not None, f"The 'name' field is missing in {x}"
+            data = x.get("data", None)
+            if data is not None:
+                assert data is not None, f"The 'data' field is missing in {x}"
+                file_name = utils.decode_base64_to_file(data, dir=save_dir).name
+            if x.get("is_file", True):
+                filepath = x.get("path")
+                assert filepath is not None, f"The 'path' field is missing in {x}"
                 file_name = utils.download_file(
                     root_url + "file=" + filepath,
                     hf_token=hf_token,
                     dir=save_dir,
                 )
```

Comment:
I swapped this around because we don't have `is_file` any more, `is_file` is implicit. For previous version i think the `data` check should work, if it is present it should be base64 data. New versions that do not have a data key will skip this branch.

Reply:
@freddyaboulton does that mean I can get rid of this too?

Reply:
Sorry I missed this - what do you mean?

Reply:
This is for backwards compat too, was referring to the above comment as backwards compat being handled by another class.